### ISSUE TITLE
`NSString.lines()` generates surplus line when string ended with new line character

### ISSUE THREAD

**Issue by norio-nomura** [jpsim/SourceKitten#259]
This is introduced by #254 and causes [test failures](https://travis-ci.org/realm/SwiftLint/jobs/162080744#L761) of https://github.com/realm/SwiftLint/pull/771.

**Comment by norio-nomura** [comment]
Or previous behavior was wrong?

**Comment by norio-nomura** [comment]
`WC(1)`'s behavior is following:
```sh
$ head -c 100 < /dev/zero | tr '\0' '\n' | wc -l
     100
```

**Comment by norio-nomura** [comment]
```console
$ TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 swift
Welcome to Apple Swift version 2.3 (swiftlang-800.10.12 clang-800.0.38). Type :help for assistance.
  1> import Foundation 
  2> Repeat(count: 100, repeatedValue: "\n") 
  3.   .joinWithSeparator("") 
  4.   .componentsSeparatedByCharactersInSet(.newlineCharacterSet()).count 
$R0: Int = 101
```

**Comment by norio-nomura** [comment]
Xcode displays `\n` only file as containing 2 lines:
<img width="712" alt="screenshot 2016-09-24 18 28 11" src="https://cloud.githubusercontent.com/assets/33430/18807529/b2377214-8284-11e6-8967-07d7e80e3eae.png">

**Comment by norio-nomura** [comment]
Ok, we'll go with behavior that same as we did before #254.
Fixed by #258.
